File: src/test/java/TwoSignOneImplTest.java
public class TwoSignOneImplTest {
}

interface X {

}

interface Y {

}

class Z implements X, Y {

}

interface A<T extends X> {
    public void foo(T item);
}

class C implements A<Z> {

    @Override
    public void foo(Z item) {

    }
}

File: src/test/kotlin/com/npcode/learning/kotlin/TwoSignOneImplTest.kt
package com.npcode.learning.kotlin

class TwoSignOneImplTest {
}

interface X

interface Y

class Z : X, Y

interface A<T : X> {
    fun foo(item: T)
}

interface B<T : Y> {
    fun foo(item: T)
}

class C : A<Z>, B<Z> {
    override fun foo(item: Z) {
        TODO("Not yet implemented")
    }
}


